cpp/src/arrow/compute/exec/CMakeLists.txt

```diff
@@ -32,6 +32,9 @@ add_arrow_compute_test(util_test PREFIX "arrow-compute")
 
 add_arrow_benchmark(expression_benchmark PREFIX "arrow-compute")
 
+find_package(OpenMP)
+add_arrow_benchmark(hash_join_benchmark PREFIX "arrow-compute" EXTRA_LINK_LIBS OpenMP::OpenMP_CXX)
```

Comment:
If OpenMP is not available, would benchmark still compile?

Reply:
No, probably not. I believe the goal is to be able to also compare the overheads of our executor framework vs OpenMP at some point

Reply:
I changed it so that it won't break compilation - it'll just not build this particular target in the absence of OpenMP.